x-pack/plugin/data-science/src/main/java/org/elasticsearch/xpack/datascience/DataSciencePlugin.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.datascience;
+
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.plugins.SearchPlugin;
+import org.elasticsearch.xpack.core.XPackPlugin;
+import org.elasticsearch.xpack.datascience.cumulativecardinality.CumulativeCardinalityPipelineAggregationBuilder;
+import org.elasticsearch.xpack.datascience.cumulativecardinality.CumulativeCardinalityPipelineAggregator;
+
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+
+public class DataSciencePlugin extends Plugin implements SearchPlugin {
+
+    // volatile so all threads can see changes
+    protected static volatile boolean isDataScienceAllowed;
+
+    public DataSciencePlugin() {
+        registerLicenseListener();
```

Comment:
I actually don't think all this stuff is needed, since you can't go from Basic+ to OSS without uninstalling/reinstalling (I think).  So there's no need to watch and cache license changes for a Basic+ feature here.

Will verify and adjust accordingly.

Reply:
Looks like it's still needed, at least until https://github.com/elastic/elasticsearch/issues/45022 is resolved.